src/DaedalusCompiler/Dat/DatSymbol.cs

```diff
@@ -107,7 +103,9 @@ public class DatSymbol
         /// Reference to parent symbol for nested symbols like class variables
         /// </summary>
         public int ParentIndex { get; set; }
-
+        
+        public ErrorLineContext ErrorLineContext;
```

Comment:
Why symbol need that `ErrorLineContext ` field?

Reply:
Because I would like to remember context that symbol was in. For example with this kind of error:
```
test.d:6:4: error: too many arguments to function call, expected 1, have 2
    testFunc(2, 3);
    ^
test.d:1:9: note: 'testFunc' declared here
func int testFunc(var int a) {};
         ^
```
I needed `testFunc` symbol's ErrorLineContext to display note showing where the symbol occured. This kind of note output is in CLang compiler and I thought it would be good idea to have it. Can be really helpful to tell people where the function is declared.